### PR TITLE
Add support for Seurat v3

### PR DESCRIPTION
**Pull request by mojaveazure**
Hi Scott,

MAGIC is really cool, I like how there's native support for Seurat objects in it. I've added support for the Seurat v3 object (currently available as a [prerelease](https://github.com/satijalab/seurat/tree/release/3.0)) as it currently only supports the v2 object. I've done this by converting `magic` to an S3 generic and method; instead of the function trying to figure out what object was passed to it and how to extract the data, R now takes care of that.

There are now `magic.seurat` for v2 objects and `magic.Seurat` for v3 objects. Each of these calls `magic.default` once they extract the data they need, and add it back in the correct manner. The user sees no difference, as they just call `magic` and R takes care of the rest.

This does depend on the eventual CRAN release of Seurat v3, so hopefully I'm not messing with any release schedule you had planned.

### CODE REVIEW

**Comment by scottgigante** [comment]
Hi @mojaveazure ,

This is fantastic! Thank you very much. What is your release schedule for Seuratv3?

**Comment by mojaveazure** [comment]
Hi Scott,

We're currently reworking our documentation and vignettes, as well as working out some newly-found edge-case bugs. We'll keep you posted on when we're preparing our release.

**Comment by scottgigante** [comment]
Great, thanks. A side note: the CI build is currently failing because it appears the changes you added made Seurat a dependency for Rmagic. Is it possible to add these S3 methods without adding Seurat as a dependency?

File: Rmagic/R/magic.R [hunk 1]
```diff
@@ -269,11 +267,14 @@ magic.Seurat <- function(
   seed = NULL,
   ...
 ) {
+  if (!requireNamespace(pacakge = 'Seurat', quietly = TRUE)) {
```
Comment (scottgigante) [inline Rmagic/R/magic.R]: There's a typo here: pacakge -> package

**Comment by scottgigante** [comment]
This is great, thanks @mojaveazure ! Do you mind adding Seurat to the .travis.yml so that the CI can run successfully?

**Comment by scottgigante** [comment]
Thanks so much! Merged

**Pull request merged**